### ISSUE TITLE
Failed initial INFO message from server is hard to understand

### ISSUE THREAD

**Issue by AlbericC** [nats-io/nats.go#1103]
## Defect

When connecting to a public NATS server from behind an WAF  (web application firewall) and through TLS, the unencrypted INFO initial message can be caught by the WAF, or truncated. 

In this case the `(*Conn).readOp` in  `(*Conn).processExpectedInfo()`  may read an empty buffer, later on leading to json.Unmarshal in `(*Conn)processInfo(c.args)` to return and error that simply reads "EOF".

In these cases it is difficult to diagnose what happens, especially since the error is returned "naked" all the way through 

https://github.com/nats-io/nats.go/blob/543e628f4bafe989ce2ba548af09e2067c67ea69/nats.go#L3221

https://github.com/nats-io/nats.go/blob/543e628f4bafe989ce2ba548af09e2067c67ea69/nats.go#L2220

https://github.com/nats-io/nats.go/blob/543e628f4bafe989ce2ba548af09e2067c67ea69/nats.go#L2088

https://github.com/nats-io/nats.go/blob/543e628f4bafe989ce2ba548af09e2067c67ea69/nats.go#L2135

https://github.com/nats-io/nats.go/blob/543e628f4bafe989ce2ba548af09e2067c67ea69/nats.go#L2175

https://github.com/nats-io/nats.go/blob/543e628f4bafe989ce2ba548af09e2067c67ea69/nats.go#L1369

https://github.com/nats-io/nats.go/blob/543e628f4bafe989ce2ba548af09e2067c67ea69/nats.go#L784

Make sure that these boxes are checked before submitting your issue -- thank you!

 - [X] Included nats.go version
 - [ ] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve)

#### Versions of `nats.go` and the `nats-server` if one was involved:
nats.go 1.17.0

#### Steps or code to reproduce the issue:

I honestly don't know how to reproduce it without a WAF. 

#### Expected result:

A transmission error stating the received initial INFO message from the server is invalid

#### Actual result:

"nats: error: EOF"

![unclear-error](https://user-images.githubusercontent.com/7861071/194722572-3e8d08f0-f09f-4cd8-a54a-d168ef15b7a9.png)

#####Side note :
the WAF may have been triggered because the NATS system we are trying to reach is exposed on port 443, in a (not so smart) attempt to use a well-known port, so that firewall would likely let the connection pass though.

**Comment by derekcollison** [comment]
The Go client supports using the websocket protocol and the server can talk native websockets as well. The demo server has this enabled and should work with WAF.

**Comment by AlbericC** [comment]
I should have pointed that the case where those errors happened is a regular NATS with TLS enabled, except running on port 443

**Comment by derekcollison** [comment]
Yes understood, the client expects a plaintext INFO block and when it does not see it or the socket gets chopped the only reasonable thing to report seems to be EOF.
